internal/runtimehandlerhooks/runtime_handler_hooks.go

```diff
@@ -16,39 +17,26 @@ type RuntimeHandlerHooks interface {
 }
 
 // GetRuntimeHandlerHooks returns RuntimeHandlerHooks implementation by the runtime handler name
-func GetRuntimeHandlerHooks(ctx context.Context, config *libconfig.Config, handler string, r *oci.Runtime) (RuntimeHandlerHooks, error) {
-	allowAnnotations, err := allowHighPerformanceAnnotations(handler, r)
-	if err != nil {
-		return nil, err
-	}
-
-	if allowAnnotations {
+func GetRuntimeHandlerHooks(ctx context.Context, config *libconfig.Config, handler string, annotations map[string]string) (RuntimeHandlerHooks, error) {
+	if strings.Contains(handler, HighPerformance) {
+		log.Warnf(ctx, "The usage of the handler %q without adding high-performance feature annotations under allowed_annotations will be deprecated under 1.21", HighPerformance)
 		return &HighPerformanceHooks{config.IrqBalanceConfigFile}, nil
 	}
-
-	if !allowAnnotations && strings.Contains(handler, HighPerformance) {
+	if highPerformanceAnnotationsSpecified(annotations) {
```

Comment:
I think it should be `!highPerformanceAnnotationsSpecified`

Reply:
I actually dropped that case, as we're preparing for 1.21 anyway